src/object_store/src/object/mod.rs

```diff
@@ -273,8 +273,8 @@ fn try_update_failure_metric<T>(
     result: &ObjectResult<T>,
     operation_type: &'static str,
 ) {
-    if let Err(e) = &result {
-        tracing::error!("{:?} failed because of: {:?}", operation_type, e);
+    if let Err(e) = &result && !e.is_object_not_found_error() {
```

Comment:
hmm.... For backup this may be expected, but for state store, object not found is actually a severe issue so failure metric and logging are both useful. Is it possible to prevent backup from constantly issuing read on non-existent objects instead? Or we can check the object suffix to blacklist error reporting for backup manifest file.

Reply:
> prevent backup from constantly issuing read on non-existent objects instead
It only issues once on startup.

> check the object suffix to blacklist error reporting for backup manifest file.
Will fix.